Renovate the docs

Lunarnovaa opened the pull request:
General PR to address some current issues of the docs. Namely:

- [x] Improve general cohesion of documentation
- [x] Add documentation to our docs page
- [x] Fix the custom link (CNAME)
- [ ] See if I can mess with CSS to add syntax highlighting and maybe change up the color scheme to be more individual to HJR
- [ ] Clean up the docs more generally, unify tone and such
- [x] Break up CONTRIBUTING.md into multiple smaller files that link together (and modify the sidebar to make this cleaner)
- [x] Add a simple index.md landing page

### Current Blockers:

- [x] New SNUG website domain
- [x] NDG additive stylesheet option

### Meta
Related Issue: #73

[We do not currently have a policy against AI-generated code, but we ask you to disclose the usage of AI for code generation]::
AI used to generate code included in this PR?: No


### All Submissions:
 - [x] Formatted commit message in accordance with CONTRIBUTING.md guidelines
 - [x] Filled in all meta items
 - [x] Mentioned any blockers before the PR can merge
 - [x] Verified there are no conflicting PRs open

nezia1 reviewed (dismissed):
Fantastic work. LGTM, added a review comment but non blocking

Lunarnovaa commented:
- Moved CNAME to own file
- Moved markdown files to docs/manual
- Small cleanup/refactor

Lunarnovaa commented:
Adding a couple more things.

Lunarnovaa commented:
Addressed nezia's request and added a workflow to test build the docs. Please give feedback on the workflow, I'm pretty unfamiliar with them.

nezia1 reviewed (commented):
One last question, but LGTM otherwise. The section on `defaultText` is really good stuff :)

Lunarnovaa commented:
- Docs check workflow no longer runs if only tests were changed
- Example test is now nushell, not ncmpcpp
- Added link checking to the docs check workflow

Lunarnovaa commented:
The 29 errors on the link checker workflow is a bug with ndg atm, should be fine on our actual docs.

nezia1 commented:
LGTM, but holding my review until ndg's bug is fixed

Lunarnovaa commented:
Alright, so the errors aren't a _bug_ perse with ndg, rather, it's due to the javascript that adds anchors to headings not actually running at buildtime, even though it totally works fine on the published site. So, I've added anchors to the headings manually, and updated CONTRIBUTING to reflect that.

Lunarnovaa commented:
Superseded by #115.
